src/Controller/Model/CategoriesController.php

```diff
@@ -39,21 +38,50 @@ class CategoriesController extends ModelBaseController
     protected $singleView = false;
 
     /**
-     * {@inheritDoc}
+     * @inheritDoc
      */
-    public function beforeRender(Event $event): ?Response
+    public function index(): ?Response
     {
-        $features = $this->Schema->objectTypesFeatures();
-        $categorized = (array)Hash::get($features, 'categorized');
-        $schema = $this->Schema->getSchema();
-        $schema['properties']['type']['enum'] = $categorized;
-        $filter = (array)$this->request->getQuery('filter');
-        if (!empty($filter['type'])) {
-            $categorized = [$filter['type']];
+        $this->request->allowMethod(['get']);
+        $query = $this->request->getQueryParams() + [
+            'page_size' => 500,
+        ];
+
+        try {
+            $response = $this->apiClient->get('/model/categories', $query);
```

Comment:
500 categories could be all, or not. 
you could do something as follows:
```
$query = $this->request->getQueryParams() + [
    'page_size' => 500,
];
try {
    $response = $this->apiClient->get('/model/categories', $query);
    $data = (array)$response['data'];
    $pageCount = $response['meta']['pagination']['page_count'];
    if ($pageCount > 1) {
        for ($i = 1; $i < $pageCount; $i++) {
            $query['page'] = $i + 1;
            $response = $this->apiClient->get('/model/categories', $query);
            $data = array_merge($data, (array)$response['data']);
        }
    }
}
// then use $data instead of $response['data']
```

Reply:
Categories index is paginated, we don't need to load all categories at once. Anyway, 500 is not a valid value for BEdita because the maximum is 100 I think. I'll fix it.